tools/themis/src/rules.rs

```diff
@@ -0,0 +1,351 @@
+pub fn has_debuggable_rust_version(workspace: &Workspace) -> Result<(), Error> {
+    let rust_toolchain =
+        utils::parse_toml::<toml::Value>(&workspace.root.join("rust-toolchain.toml"))?;
+    let rust_toolchain = rust_toolchain["toolchain"]["channel"].as_str().unwrap().to_owned();
+
+    let rust_toolchain = match semver::Version::parse(&rust_toolchain) {
+        Ok(rust_toolchain) => rust_toolchain,
+        Err(err) => {
+            utils::warn!(
+                "semver: unable to parse rustup channel from {}: {}",
+                style::highlight("rust-toolchain.toml"),
+                err
+            );
+
+            return Ok(());
+        }
+    };
+
+    let mut outliers = vec![];
+    for pkg in &workspace.members {
+        let (rust_version, raw) = match pkg.raw["package"].get("rust-version") {
+            Some(rust_version) => {
+                let raw = rust_version.as_str().unwrap();
+                (semver::VersionReq::parse(raw)?, raw)
+            }
+            // we can skip, since we have has_rust_version check
+            None => continue,
+        };
+
+        if !rust_version.matches(&rust_toolchain) {
+            outliers.push(PackageOutcome { pkg, value: Some(raw.to_owned()) });
+        }
+    }
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages have an incompatible `rust-version`".to_string(),
+            expected: Some(Expected {
+                value: format!("<={}", rust_toolchain),
+                reason: Some("as defined in the `rust-toolchain`".to_string()),
+            }),
+            outliers,
+        });
+    }
+
+    return Ok(());
+}
+
+/// ensure all crates share the same rust edition
+pub fn has_unified_rust_edition(workspace: &Workspace) -> Result<(), Error> {
+    let mut edition_groups = HashMap::new();
+
+    for pkg in &workspace.members {
+        *edition_groups.entry(&pkg.parsed.edition).or_insert(0) += 1;
+    }
+
+    let (most_common_edition, n_compliant) =
+        edition_groups.into_iter().reduce(|a, b| if a.1 > b.1 { a } else { b }).unwrap();
+
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.parsed.edition != *most_common_edition)
+        .map(|pkg| PackageOutcome { pkg, value: Some(pkg.parsed.edition.clone()) })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages have an unexpected rust edition".to_string(),
+            expected: Some(Expected {
+                value: most_common_edition.clone(),
+                reason: Some(format!("used by {} other packages in the workspace", n_compliant)),
+            }),
+            outliers,
+        });
+    }
+
+    return Ok(());
+}
+
+/// ensure all crates have the appropriate author, non-exclusively of course.
+pub fn author_is<'a>(expected: &'a str) -> impl Fn(&'a Workspace) -> Result<(), Error> {
+    move |workspace| {
```

Comment:
Not a huge fan of high-order functions. I'd just hard-code the author string.

Reply:
Artifacts of, like you mentioned, a more generic design. I'll resolve it.